test/util/util.go

```diff
@@ -130,11 +137,34 @@ func WaitForInstanceCondition(client v1alpha1servicecatalog.ServicecatalogV1alph
 	)
 }
 
+// WaitForInstanceToNotExist waits for the Instance with the given name to no
+// longer exist.
+func WaitForInstanceToNotExist(client v1alpha1servicecatalog.ServicecatalogV1alpha1Interface, namespace, name string) error {
+	return wait.PollImmediate(500*time.Millisecond, wait.ForeverTestTimeout,
+		func() (bool, error) {
+			glog.V(5).Infof("Waiting for instance %v/%v to not exist", namespace, name)
+
+			_, err := client.Instances(namespace).Get(name)
+			if nil == err {
+				return false, nil
+			}
+
+			if errors.IsNotFound(err) {
+				return true, nil
+			}
```

Comment:
I know it's unlikely, but I'd like to be defensive here and below and check for all other `err != nil` scenarios